### ISSUE TITLE
Твик и ссылки

### ISSUE THREAD

**Issue by ukich-hubber** [anilibria/anilibria-ios#33]
Появилась интересная идея на счет добавления в приложения твика для открытия ссылок сразу в приложении. Как мне кажется довольно таки удобно было бы сделать твик который бы позволял сразу открыть аниме в приложении(скриншот приложу ниже), то есть то самое небольшое всплывающие окошко которое предлагало бы открывать аниме в приложении . А также хотелось бы чтобы снова была возможность копировать и отправлять ссылки на тайтлы прямо из приложения, как это было в прошлой версии
![image](https://github.com/user-attachments/assets/a7badb27-9126-4ba1-b809-b2fce05ceedb)

**Comment by Allui** [comment]
> копировать и отправлять ссылки на тайтлы прямо из приложения, как это было в прошлой версии

Ля пока переносил на новое апи временно убрал и совсем про это забыл.

> Появилась интересная идея на счет добавления в приложения твика для открытия ссылок сразу в приложении.

А вот открыть ссылку прямо в приложении к сожалению не получится. Это доступно только для официальных приложений как собственно и пуш уведомления.

**Comment by ukich-hubber** [comment]
Насчет твиков по моему это как то реализуется, просто также скачивал сторонние модифицированные приложения ipa и для некоторых из них есть твики реализованные через расширение в браузере (фото приложу чуть ниже), но если это все же невозможно, то конечно жаль

**Comment by ukich-hubber** [comment]
А, и забыл на счет уведомлений. В некоторых неофициальных приложениях они также реализованы (например Hdrezka, ниже приложу скрин), конечно не знаю как, но сам факт что как то можно

**Comment by hvitalikh9** [comment]
Согласен, разработчики приложения hdrezka как то смогли сделать уведомления, ниже есть скрины 
![IMG_4926](https://github.com/user-attachments/assets/36e2430b-70bf-4c1c-a3de-de4084c047aa)
![IMG_4927](https://github.com/user-attachments/assets/1306f55f-c593-4fa6-9d42-90aebca771a1)

**Comment by ukich-hubber** [comment]
**Дополняю свой пост небольшой информацией.**
**Уведомления**
Все же в неофициальном приложении можно добавить уведомления если есть аккаунт разработчика стоимостью 100$/год, однако если его нету то уведомления также можно сделать, но их нельзя будет связать с серверными функциями (например уведомление о выходе новой серии в тайтле), будут работать только локальные уведомления по типу завершение загрузки серии через торрент в приложении (пока нет в приложении, но это самый понятный пример)
**Ссылки**
На счет ссылок, можно также сделать ссылку для открытия в приложении с помощью создания url type и после в руте onOpenUrl. Опять же это будет локальная ссылка (то есть не обычная ссылка на тайтл, а отдельная ссылка, непохожая на обычную, с путем например "anl://anime/76912-fairy-tail", такая ссылка будет подходить только к приложению на iOS. Как идея сделать возможность выбора какой ссылкой поделиться (ниже приложу скрин)![image](https://github.com/user-attachments/assets/9971c880-a068-485b-be27-785501ed1ead)
Где "Сайт" это общая ссылка ведущая на страницу тайтла на сайт анилибрии,
А "Универсальная ссылка" то есть deeplink, открывала бы ссылку сразу в приложении.
**Подытог**
Как такого смысла от уведомлений сделанных без сертификата разработчика (локальные) нету, возможно в будущем если у вас появится сертификат, то можно будет реализовать уведомления связанные с тайтлами и остальным (серверными), но пока как есть
А вот возможность сделать сделать deeplink для открытия тайтла сразу в приложении можно. 
Большая просьба написать вас ваше мнение на этот счет. 
_Заранее спасибо за ответ. _

**Comment by Allui** [comment]
> Все же в неофициальном приложении можно добавить уведомления если есть аккаунт разработчика стоимостью 100$/год

Я конечно точно не помню, но серверные приложения вроде не должны работать так как существующие методы установки подменяют сертификат разработчика, на сертификат пользователя, а серверные уведомления могут работать только с оригинальным сертификатом (разработчика) 🤔

**Comment by maksim-katafalk** [comment]
Чтобы в самом приложении было наличие уведомлений надо чтобы вы сделали их с помощью сертификата, а чтобы они работали у обычных юзеров, надо чтобы у них был сертификат на своем устройстве, то есть если установить приложение через altstore, то уведомления работать не будут. 
Но сейчас большая часть всех пользователей используют личные сертификаты на своих устройствах, тк появилось довольно таки много компаний которые продают сертификаты довольно таки дешево.
Так что по моему мнению если есть возможность, то обязательно стоит их сделать, тк для большинства юзеров это будет очень полезно.

**Comment by Allui** [comment]
> если установить приложение через altstore, то уведомления работать не будут.
Но сейчас большая часть всех пользователей используют личные сертификаты на своих устройствах, тк появилось довольно таки много компаний которые продают сертификаты с которыми работают уведомления довольно таки дешево.

Хотите сказать что если подписать приложение используя сертификат другого разработчика, то заработают серверные уведомления? Каким это образом? Сертификаты же не будут совпадать. Или я что то упускаю?🤔

**Comment by maksim-katafalk** [comment]
Скорее всего вы думаете о чем то другом, потому что имея любой официальный сертификат от apple на устройстве можно подписать приложение и будут работать уведомления, если хотите могу записать видео где будет показано что оно работает
